restapi/src/main/java/io/stargate/web/docsapi/service/ReactiveDocumentService.java

```diff
@@ -110,6 +110,39 @@ public ReactiveDocumentService(
     this.configuration = configuration;
   }
 
+  private Integer determineTtl(
+      String ttlRaw,
+      DocumentDB db,
+      String namespace,
+      String collection,
+      String documentId,
+      List<String> subPath)
+      throws UnauthorizedException {
+    // ttlRaw is either an integer, a String equaling 'auto', or null
+    if (ttlRaw == null) {
+      return null;
+    }
+    try {
+      Integer ttlValue = Integer.parseInt(ttlRaw);
+      if (subPath.isEmpty()) {
+        return ttlValue;
+      } else {
+        throw new ErrorCodeRuntimeException(ErrorCode.DOCS_API_INVALID_TTL_NUMBER);
+      }
+    } catch (NumberFormatException e) {
+      if (!subPath.isEmpty()) {
+        if (!ttlRaw.equalsIgnoreCase("auto")) {
```

Comment:
I would suggest explicitly looking for "auto" first since exceptions are rather expensive things to throw -- so creating and catching `NumberFormatException`s can be wasteful. More efficient to check for special case(s) first.
Same goes for empty String actually, probably better to check those 2 before attempting number parsing.